.github/workflows/publish_plugins.yml

```diff
@@ -0,0 +1,63 @@
+name: Publish Built-in Plugins
+on:
+  release:
+    types:
+      - published
+
+jobs:
+  publish:
+    name: Publish plugins
+    runs-on: ubuntu-latest
+    env:
+      SCARB_REGISTRY_AUTH_TOKEN: ${{ secrets.SCARB_REGISTRY_AUTH_TOKEN }}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: software-mansion/setup-scarb@v1
+        with:
+          scarb-version: ${{ github.ref_name }}
+
+      - name: Init Scarb to cache plugins
+        id: scarb-init
+        run: |
+          set -eo pipefail
+          mkdir -p $HOME/pkg
+          pushd $HOME/pkg
+          scarb init --no-vcs
+          echo "CAIRO_VERSION=$(scarb metadata --format-version 1 | jq -r '.app_version_info.cairo.version')" >> $GITHUB_OUTPUT
+        env:
+          SCARB_INIT_TEST_RUNNER: cairo-test
+
+      - name: Check if plugin versions exist in the registry
```

Comment:
Why do we need to check it here? Can't we just try to publish and let the pipeline fail if it has already been uploaded? (Generally it expect it not to be, right?)

Reply:
> (Generally it expect it not to be, right?)

Since some Cairo versions correspond to multiple Scarb versions, this is likely to happen from time to time. 

> Can't we just try to publish and let the pipeline fail if it has already been uploaded?

I think this is the question of deciding how do we want the action to resolve in such instances. We can `continue-on-error`, and let entire workflow fail, or we can skip the uploads. It seems to me that if action is resolved within *what is expected* should not be deemed "failed".

Reply:
Makes sense, thanks!